MoreLinq.Test/FallbackIfEmptyTest.cs

```diff
@@ -39,6 +39,28 @@ public void FallbackIfEmptyWithEmptySequence()
         }
 
         [Test]
+        public void FallbackIfEmptyPreservesSourceCollectionIfPossible()
+        {
+            var source = new int[] { 1 };
+            // ReSharper disable PossibleMultipleEnumeration
+            Assert.AreSame(source.FallbackIfEmpty(12), source);
```

Comment:
But those test cases don't add much value, as they all test the same overload. The idea is to test that all overloads preserve the source collection.